### DISCUSSION TITLE
How do you back up ejbca if you're using the Docker image?

### DISCUSSION THREAD

**Discussion by stevesobol** [Keyfactor/ejbca-ce#31] [Q&A] [answered]
Hello,

Backing up EJBCA seems straightforward if you're not using Docker. But I am running EJBCA using [the official Docker image.](https://hub.docker.com/r/primekey/ejbca-ce) How do I perform backups if using Docker? Thanks.

**Comment by primetomas** [comment, marked as answer]
Using the quick start will use a H2 database built into WildFly. It is not suitable for production. For production use an external database, like a MariaDB container and connect to that (docs in dockerhub). Then do a nice backup of the MariaDB database.

Cheers,
Tomas

**Comment by stevesobol** [comment]
Yes, I'm planning on setting up a MariaDB database. Is that the only thing I need to back up? I'd also like to back up whatever files are used to configure EJBCA.

Thanks :)

**Comment by primetomas** [comment]
The good thin is that all operational data is i the database. So by backing up the database you can disaster recover the CA system. Especially if using the container, when you can just start a new container pointing to the same database (you can also create a cluster this way).
Other things that may be "convenience" backed up are TLS certificates and trust stores (acceptable CA certificates for TLS). These are inside the container I guess, but can also easily be re-created. The "container" itself should be possible to have ephemeral as long as the data resides safely in the database.
